### ISSUE TITLE
Non-AD: Add GitHub action for building Windows package on tag builds

### ISSUE THREAD

**Issue by zmb3** [gravitational/teleport#18994]
Given that we're moving away from Drone, let's use GHA to build releases of the Windows package.

**Comment by wadells** [comment]
This isn't quite done yet, as there is some discussion ongoing about how these bits get published.  I got involved when a request to reuse a secret that Drone uses to publish the main teleport bits crossed my desk.

Particularly, it sounds like we decided to use the Cloud's release service.  However, this choice comes with issues:

https://github.com/gravitational/cloud/issues/2223

TLDR: the release service lacks sandboxing that could stop the code in this repo from e.g. publishing and promoting a new Teleport release.  As such, if we want to give a release secret to GitHub Actions in `teleport-windows-auth` repo, we need to take some extra steps:

* Add a new user to platform.teleport.sh and provision new secrets for this GHA integration (machine ID would be nice here, but I don't think we're quite there yet, so 3 month certs it is).  This allows us to track which repo is making which api calls, and lock one without impacting the other.
* Harden the `teleport-windows-auth` repo to the standards of our other production repos.  This keeps the high value release secrets safe, so that they can't be exfiltrated any more easily than they could from the Teleport repo.

I guess my bigger question is: Why isn't this this project of the regular teleport repo and build?  Is there any documented decision making behind this choice (a RFD or slack/issues discussion)?

We're already planning to consolidate teleport-plugins into teleport and have been moving other infra in.  I'm concerned that this is an important part of the product that is circumventing our typical standards and processes.
